ludwig/utils/defaults.py

```diff
@@ -16,357 +16,38 @@
 import argparse
 import copy
 import logging
-import sys
-from dataclasses import asdict
-from typing import Any, Dict
 
 import yaml
 
-from ludwig.constants import (
-    COLUMN,
-    COMBINER,
-    DECODER,
-    DEFAULTS,
-    DROP_ROW,
-    ENCODER,
-    EXECUTOR,
-    HYPEROPT,
-    INPUT_FEATURES,
-    LOSS,
-    MODEL_ECD,
-    MODEL_GBM,
-    MODEL_TYPE,
-    NAME,
-    OUTPUT_FEATURES,
-    PREPROCESSING,
-    PROC_COLUMN,
-    RAY,
-    SPLIT,
-    TRAINER,
-    TYPE,
-)
 from ludwig.contrib import add_contrib_callback_args
-from ludwig.data.split import DEFAULT_PROBABILITIES, get_splitter
-from ludwig.features.feature_registries import base_type_registry, input_type_registry, output_type_registry
-from ludwig.features.feature_utils import compute_feature_hash
+from ludwig.features.feature_registries import input_type_registry
 from ludwig.globals import LUDWIG_VERSION
-from ludwig.schema.combiners.utils import combiner_registry
-from ludwig.schema.utils import load_config_with_kwargs, load_trainer_with_kwargs
-from ludwig.utils.config_utils import get_default_encoder_or_decoder, get_defaults_section_for_feature_type
+from ludwig.schema import validate_config
+from ludwig.schema.config_object import Config
+from ludwig.schema.preprocessing import PreprocessingConfig
+from ludwig.utils.backward_compatibility import upgrade_to_latest_version
 from ludwig.utils.data_utils import load_config_from_str, load_yaml
 from ludwig.utils.fs_utils import open_file
-from ludwig.utils.misc_utils import get_from_registry, merge_dict, set_default_value, set_default_values
 from ludwig.utils.print_utils import print_ludwig
 
 logger = logging.getLogger(__name__)
 
 default_random_seed = 42
 
-BASE_PREPROCESSING_SPLIT_CONFIG = {"type": "random", "probabilities": list(DEFAULT_PROBABILITIES)}
-base_preprocessing_parameters = {
-    "split": BASE_PREPROCESSING_SPLIT_CONFIG,
-    "undersample_majority": None,
-    "oversample_minority": None,
-    "sample_ratio": 1.0,
-}
-
+# Still needed for preprocessing  TODO(Connor): Refactor ludwig/data/preprocessing to use schema
 default_feature_specific_preprocessing_parameters = {
-    name: base_type.preprocessing_defaults() for name, base_type in base_type_registry.items()
+    name: preproc_sect.get_schema_cls()().preprocessing.to_dict() for name, preproc_sect in input_type_registry.items()
 }
 
 default_preprocessing_parameters = copy.deepcopy(default_feature_specific_preprocessing_parameters)
-default_preprocessing_parameters.update(base_preprocessing_parameters)
```

Comment:
👏👏👏 No action item here just happy to see so much defaults.py deleted.

Reply:
Thank you:)